⛔️ Add better error messages as rest response

daniel-rose opened the pull request:
- effected bundle: fond-of-oryx/splittable-checkout-rest-api
 - use checkout error object for returning better error message
 - prepare unit tests
 - ...

julianzimmermann commented on bundles/erp-order-api/src/FondOfOryx/Zed/ErpOrderApi/Business/Model/ErpOrderApi.php:
```diff
@@ -0,0 +1,154 @@
+
+        if ($erpOrderTransfer === null || !$erpOrderResponseTransfer->getIsSuccessful()) {
+            throw new EntityNotSavedException(
+                sprintf('Could not create erp order.'),
```
sprintf('Could not create erp order. Error code %s', ApiConfig::HTTP_CODE_INTERNAL_ERROR)

julianzimmermann commented on bundles/erp-order-api/src/FondOfOryx/Zed/ErpOrderApi/Business/Model/ErpOrderApi.php:
```diff
@@ -0,0 +1,154 @@
+
+        if ($erpOrderTransfer === null || !$erpOrderResponseTransfer->getIsSuccessful()) {
+            throw new EntityNotSavedException(
+                sprintf('Could not update erp order.'),
```
sprintf('Could not update erp order. Error code %s', ApiConfig::HTTP_CODE_INTERNAL_ERROR)

julianzimmermann commented on bundles/erp-order-api/src/FondOfOryx/Zed/ErpOrderApi/Business/Model/ErpOrderApi.php:
```diff
@@ -0,0 +1,154 @@
+
+        if ($erpOrderTransfer === null) {
+            throw new EntityNotFoundException(
+                sprintf('Could not found erp order.'),
```
sprintf('Could not find erp order with id %s. Error code %s', idErpOrder, ApiConfig:: HTTP_CODE_NOT_FOUND)

julianzimmermann reviewed (commented):
see comments

daniel-rose committed:
⛔️ Add better error messages as rest response
 - effected bundle: fond-of-oryx/splittable-checkout-rest-api
 - use checkout error object for returning better error message
 - prepare unit tests
 - ...

daniel-rose committed:
WIP
